### ISSUE TITLE
Update `unavailable_function` rule to also detect `preconditionFailure()` and `abort()`

### ISSUE THREAD

**Issue by quinntaylor** [realm/SwiftLint#3374]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The `unavailable_function` rule warns when a function contains `fatalError()` with no other return branches, but does not detect `preconditionFailure()` or `abort()` calls, which also terminate execution.

Although Xcode fix-its often suggest `fatalError()`, some developers find it preferable to use `preconditionFailure()` instead.

https://lapcatsoftware.com/articles/fatalError.html

### Environment

* SwiftLint version: 0.38.2
* Installation method used: macOS installer package
* Paste your configuration file: 

```yml
opt_in_rules:
    - unavailable_function
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using? 12.0
* Do you have a sample that shows the issue? Yes

```swift
// This does NOT trigger a violation, but should:
class ViewController: UIViewController {
    public required init?(coder aDecoder: NSCoder) {
        preconditionFailure("init(coder:) has not been implemented")
    }
}
```

**Comment by quinntaylor** [comment]
https://github.com/realm/SwiftLint/pull/3375

**Comment by quinntaylor** [comment]
@jpsim / @marcelofabri, is it possible to get some eyes on this issue and PR? Thanks!
